src/components/Profile/ProfileInfo/ProfileStatus.js

```diff
@@ -1,58 +1,42 @@
-import React, { Component } from 'react';
-
-class ProfileStatus extends Component {
-    state = {
-      editMode: false,
-      status: this.props.status,
-    };
-
-    activateEditMode = () => {
-      this.setState({
-        editMode: true,
-      });
-    };
-
-    deactivateEditMode = () => {
-      this.setState({
-        editMode: false,
-      });
-
-      this.props.updateStatus(this.state.status);
-    };
-
-    onStatusChange = (e) => {
-      this.setState({
-        status: e.currentTarget.value,
-      });
-    };
-
-    getNotEditMode = () => (
-      <div>
-        <span onDoubleClick={this.activateEditMode}>{this.props.status}</span>
-      </div>
-    );
-
-    getEditMode = () => (
-      <div>
-        <input autoFocus onChange={this.onStatusChange} onBlur={this.deactivateEditMode} value={this.state.status} />
-      </div>
-    );
-
-    componentDidUpdate(prevProps, prevState) {
-      if (prevProps.status !== this.props.status) {
-        this.setState({
-          status: this.props.status,
-        });
-      }
-    }
-
-    render() {
-      return (
-        <div>
-          {this.state.editMode ? this.getEditMode() : this.getNotEditMode()}
-        </div>
-      );
-    }
-}
+import React, { useState, useEffect } from 'react';
+
+const ProfileStatus = (props) => {
+  const [editMode, setEditMode] = useState(false);
+  const [status, setStatus] = useState(props.status);
+
+  useEffect(() => {
+    setStatus(props.status);
+  }, [props.status]);
+
+  const activateEditMode = () => {
+    setEditMode(true);
+  };
+
+  const deactivateEditMode = () => {
+    setEditMode(false);
+    props.updateStatus(status);
+  };
+
+  const onStatusChange = (e) => {
+    setStatus(e.currentTarget.value);
+  };
+
+  return (
+    <div>
+      {editMode
```

Comment:
This structure looks ugly - consider at least putting each case into a separate function.

Reply:
/fxd